sensor/common/scannerclient/grpc_client.go

```diff
@@ -2,72 +2,172 @@ package scannerclient
+// underlying scan was done through scanner-v4.
+func (i *ImageAnalysis) GetContents() *v4.Contents {
+	if i != nil {
+		return i.V4Contents
+	}
+	return nil
+}
 
-	endpoint = strings.TrimPrefix(endpoint, "https://")
-	if strings.Contains(endpoint, "://") {
-		return nil, errors.Errorf("Scanner endpoint has unsupported scheme: %s", endpoint)
+// getScannerEndpoint reads and validate the Scanner gRPC endpoint setting. If
+// the endpoint is empty or not configured properly (invalid) the value is
+// returned and error will be set.
+func getScannerEndpoint(s env.Setting) (string, error) {
+	e := s.Setting()
+	if e == "" {
+		return e, errors.Errorf("%s is not set or empty", s.EnvVar())
+	}
+	e = strings.TrimPrefix(e, "https://")
+	if strings.Contains(e, "://") {
+		return e, errors.Errorf("%s has unsupported scheme: %s", s.EnvVar(), e)
 	}
+	return e, nil
+}
 
-	tlsConfig, err := clientconn.TLSConfig(mtls.ScannerSubject, clientconn.TLSConfigOptions{
+// dial the scanner and returns a new ScannerClient.  The function is non-blocking and
+// returns a non-nil error upon configuration error.
+func dial(endpoint string, certID mtls.Subject) (*grpc.ClientConn, error) {
+	tlsConfig, err := clientconn.TLSConfig(certID, clientconn.TLSConfigOptions{
 		UseClientCert: clientconn.MustUseClientCert,
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to initialize Scanner TLS config")
+		return nil, fmt.Errorf("TLS config failed: %w", err)
 	}
-
-	// This is non-blocking. If we ever want this to block,
-	// then add the grpc.WithBlock() DialOption.
+	// This is non-blocking. If we ever want this to block, then add the
+	// grpc.WithBlock() DialOption.
+	log.Infof("dialing scanner at %s", endpoint)
 	conn, err := grpc.Dial(endpoint, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to dial Scanner")
+		return nil, fmt.Errorf("grpc dial failed: %w", err)
 	}
+	return conn, nil
+}
 
-	log.Infof("Dialing Scanner at %s", endpoint)
+// dialV2 connect to scanner V1 gRPC and return a new ScannerClient.
+func dialV2() (ScannerClient, error) {
+	endpoint, err := getScannerEndpoint(env.ScannerV4GRPCEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("dialing scanner-v2 client: %s", endpoint)
+	conn, err := dial(endpoint, mtls.ScannerSubject)
+	if err != nil {
+		return nil, err
+	}
+	return &v2Client{
+		ImageScanServiceClient: scannerV1.NewImageScanServiceClient(conn),
+		conn:                   conn,
+	}, nil
+}
 
-	return &Client{
-		client: scannerV1.NewImageScanServiceClient(conn),
-		conn:   conn,
+// dialV4 connect to scanner V4 gRPC and return a new ScannerClient.
+func dialV4() (ScannerClient, error) {
+	endpoint, err := getScannerEndpoint(env.ScannerV4GRPCEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: [ROX-19050] Set the Scanner V4 certificate here.
+	log.Infof("dialing scanner-v4 client: %s", endpoint)
+	conn, err := dial(endpoint, mtls.ScannerSubject)
```

Comment:
This is fine for now, but I think it is likely this will change when we determine how we want to make the TLS certs